ui.frontend/src/main/webpack/site/js/components/searchForm.ts

```diff
@@ -37,83 +37,106 @@ export const triggerSearch = async (
+    }
   }
 
+
   updateUrl(searchValue)
 
   removeAutosuggest(searchContainer)
   removeSearchTabs(searchContainer)
   removeSearchResults(searchContainer)
   removeSelectedTabFromSearchContainer(searchContainer)
 
-  const tabResultsArray = await Promise.all(
-    searchTabs.map(async (tab, index): Promise<Tab> => {
-      const tabResultsJSON = await fetchSearch(tab.url, searchValue)
 
-      return { ...tabResultsJSON, tabId: tab.title, index } as Tab
-    }),
-  ).finally(() => {
-    searchInputElementCopy.dataset.loading = 'false'
-  })
+  const tabResultsArray =
+      hasSearchTabs
+          ? await Promise.all(
+              searchTabs.map(async (tab, index): Promise<Tab> => {
+                const tabResultsJSON = await fetchSearch(tab.url, searchValue)
 
-  const searchFormParent = searchForm.parentElement
+                return {...tabResultsJSON, tabId: tab.title, index} as Tab
+              }),
+          ).finally(() => {
+            searchInputElementCopy.dataset.loading = 'false'
+          })
+          : null
 
-  const hasResults = tabResultsArray.some((tab) => tab.resultsTotal)
+  const searchFormParent = searchForm.parentElement
 
-  if (!hasResults) {
-    if (tabResultsArray?.[0]?.suggestion) {
-      const autoSuggestElement = buildSearchSuggestion(
-        tabResultsArray[0].suggestion.text,
-        autoSuggestText,
-      )
+  const hasResults = tabResultsArray?.some((tab) => tab.resultsTotal)
 
-      searchFormParent?.append(autoSuggestElement)
+  if (hasSearchTabs) {
+    if (hasResults) {
+      searchFormParent?.querySelector('.saas-not-found')?.remove()
     }
 
-    const notFoundElement = document.createElement('div')
-    notFoundElement.classList.add('saas-not-found')
-    notFoundElement.innerText = noResultsText
-    searchFormParent?.appendChild(notFoundElement)
+    if (!hasResults) {
+      if (tabResultsArray?.[0]?.suggestion) {
+        const autoSuggestElement = buildSearchSuggestion(
+            tabResultsArray[0].suggestion.text,
+            autoSuggestText,
+        )
 
-    return
-  }
-
-  tabResultsArray
-    .sort((tab1, tab2) => {
-      if (tab1.index < tab2.index) {
-        return 1
+        searchFormParent?.append(autoSuggestElement)
       }
 
-      if (tab2.index < tab1.index) {
-        return -1
+      if (searchFormParent) {
+        let notFoundElement = document.createElement('div')
+        notFoundElement.classList.add('saas-not-found')
+        notFoundElement.innerText = noResultsText
+        searchFormParent.appendChild(notFoundElement)
       }
 
-      return 0
-    })
-    .forEach((tabResult) => {
-      buildSearchResultsTab({
-        tabResult,
-        searchValue,
-        searchForm,
-        searchFormParent,
-        loadMoreButtonText,
-        onSearchItemClick,
-        onSwitchTab,
-        onLoadMoreButtonClick,
-        searchContainer,
-      })
-    })
+      return
+    }
+  }
```

Comment:
`if` nesting should be avoided if possible and the condition should be assigned to a variable for clarity:
`const isShowSuggestion = !hasResults &&  tabResultsArray?.[0]?.suggestion`
`const isRemoveSaasNotFound = hasSearchTabs && hasResults`
`const isDisplayNotFound = hasSearchTabs && searchFormParent`